core/textile/sync/task-executors.go

```diff
@@ -172,3 +182,48 @@ func (s *synchronizer) processBucketBackupOff(ctx context.Context, task *Task) e
 
 	return s.deleteAllFilesInPath(ctx, bucket, "")
 }
+
+func (s *synchronizer) processBucketRestoreTask(ctx context.Context, task *Task) error {
+	if err := checkTaskType(task, bucketRestoreTask); err != nil {
+		return err
+	}
+
+	bucket := task.Args[0]
+
+	if err := s.restoreBucket(ctx, bucket); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *synchronizer) processRestoreFile(ctx context.Context, task *Task) error {
+	if err := checkTaskType(task, restoreFileTask); err != nil {
+		return err
+	}
+
+	bucket := task.Args[0]
+	path := task.Args[1]
+
+	localBucket, err := s.getBucket(ctx, bucket)
+	if err != nil {
+		return err
+	}
+
+	mirrorBucket, err := s.getMirrorBucket(ctx, bucket)
+	if err != nil {
+		return err
+	}
+
+	// TODO: use timestamp or CID for check
+
+	if err = s.uploadFileToRemoteBucket(ctx, mirrorBucket, localBucket, path); err != nil {
```

Comment:
This should upload the file locally right? Maybe just a function naming thing.

Reply:
@jsonsivar 

Thank you for spotting this. Renamed in https://github.com/FleekHQ/space-daemon/pull/207/commits/2bcda3d1052a5da450071ec4301631a821e8b0ff#diff-26096467a39167c453134f075440f43a014ed45f4479be47d6ba72aaa7976829R27